File: 백준/Bronze/2798. 블랙잭/블랙잭.py
import sys
from itertools import combinations
input=lambda:sys.stdin.readline().rstrip()
n,m=map(int,input().split())
cards=[*map(int,input().split())]
max_sum=0
for combo in combinations(cards,3):
    cur_sum=sum(combo)
    if cur_sum<=m:
        max_sum=max(max_sum,cur_sum)
print(max_sum)

File: 백준/Bronze/2798. 블랙잭/블랙잭.c
#include <stdio.h>
#include <stdlib.h>
int main(void) {
    int n, m, max_sum = 0, cur_sum, *cards;
    scanf("%d %d", &n, &m);
    cards = (int*)malloc(sizeof(int) * n);
    for (int i = 0; i < n; i++)
        scanf("%d", &cards[i]);
    for (int i = 0; i < n - 2; i++) {
        for (int j = i + 1; j < n - 1; j++) {
            for (int k = j + 1; k < n; k++) {
                cur_sum = cards[i] + cards[j] + cards[k];
                if (cur_sum <= m)
                    max_sum = max_sum < cur_sum ? cur_sum : max_sum;
            }
        }
    }
    printf("%d", max_sum);
    return 0;
}